### PR TITLE
CON-1602-CON-1581-add gtm and zendesk widget

### PR DESCRIPTION
**Pull request by shomavg**

### CODE REVIEW

File: src/SFA.DAS.PSRService.Web/Views/Shared/_Layout.cshtml [hunk 1]
```diff
@@ -58,45 +58,50 @@
...
-    </environment>
+<environment exclude="Development">
+    <script src="~/lib/jquery/dist/jquery.min.js" asp-append-version="true"></script>
+    <script src="~/js/site.min.js" asp-append-version="true"></script>
+</environment>
 
-    @RenderSection("Scripts", required: false)
-    <script>if (typeof window.GOVUK === 'undefined') document.body.className = document.body.className.replace('js-enabled', '');</script>
+@RenderSection("Scripts", required: false)
+<script>if (
```
Comment (cofaulco) [inline src/SFA.DAS.PSRService.Web/Views/Shared/_Layout.cshtml]: inline script with no tag for asp-add-nonce

File: src/SFA.DAS.PSRService.Web/Startup.cs [hunk 2]
```diff
@@ -106,6 +113,76 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
...
+                    s.Self()
+                        .CustomSources("*.google-analytics.com",
+                                    "*.googleapis.com",
+                                    "*.googletagmanager.com/",
+                                   "https://www.tagmanager.google.com/",
+                                    "https://*.zdassets.com",
+                                    "https://*.zendesk.com",
+                                    "wss://*.zendesk.com",
+                                    "wss://*.zopim.com"
+                                    );
```
Comment (cofaulco) [inline src/SFA.DAS.PSRService.Web/Startup.cs]: AddNonce()

File: src/SFA.DAS.PSRService.Web/Configuration/WebConfiguration.cs [hunk 3]
```diff
@@ -24,7 +24,10 @@ public class WebConfiguration : IWebConfiguration
         public string HomeUrl { get; set; }
         [JsonRequired]
         public SessionStoreConfiguration SessionStore { get; set; }
-
         public TimeSpan? AuditWindowSize { get; set; }
+        [JsonRequired]
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/SFA.DAS.PSRService.Web/Configuration/WebConfiguration.cs]: 👍

File: src/SFA.DAS.PSRService.Web/Filters/GoogleAnalyticsFilter.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                };
+            }
+
+            base.OnActionExecuting(filterContext);
+        }
+
+        public class GaData
+        {
+            public string UserId { get; set; }
+            public string UserName { get; set; }
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/SFA.DAS.PSRService.Web/Filters/GoogleAnalyticsFilter.cs]: I have removed the UserName. But the others are being used in the _GoogleAnalyticsInfo.cshtml

File: src/SFA.DAS.PSRService.Web/Filters/ZendeskApiFilter.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                    Organization = account?.EmployerName
+                };
+            }
+
+            base.OnActionExecuting(filterContext);
+        }
+        
+        public class ZendeskApiData
+        {
+            public string UserId { get; set; }
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/SFA.DAS.PSRService.Web/Filters/ZendeskApiFilter.cs]: 👍

File: src/SFA.DAS.PSRService.Web/Filters/ZendeskApiFilter.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            public string Acc { get; set; }
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/SFA.DAS.PSRService.Web/Filters/ZendeskApiFilter.cs]: 👍

File: src/SFA.DAS.PSRService.Web/Filters/ZendeskApiFilter.cs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                    var employerAccounts =
+                        JsonConvert.DeserializeObject<Dictionary<string, EmployerIdentifier>>(
+                            employerAccountClaim
+                                ?.Value);
+
+                    if (accountIdFromUrl != null) account = employerAccounts[accountIdFromUrl];
+                }
+                controller.ViewBag.ZendeskApiData = new ZendeskApiData
+                {
+                    UserId = user.Claims.First(c => c.Type.Equals(EmployerPsrsClaims.IdamsUserIdClaimTypeIdentifier)).Value,
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/SFA.DAS.PSRService.Web/Filters/ZendeskApiFilter.cs]: 👍

File: src/SFA.DAS.PSRService.Web/Filters/ZendeskApiFilter.cs [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                    Name = user.Claims.First(c => c.Type.Equals(EmployerPsrsClaims.NameClaimsTypeIdentifier)).Value,
+                    Email = user.Claims.First(c => c.Type.Equals(EmployerPsrsClaims.EmailClaimsTypeIdentifier)).Value,
+                    Acc = account?.AccountId,
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/SFA.DAS.PSRService.Web/Filters/ZendeskApiFilter.cs]: 👍

File: src/SFA.DAS.PSRService.Web/Views/Home/Index.cshtml [hunk 9]
```diff
@@ -1,9 +1,9 @@
-﻿@using SFA.DAS.PSRService.Web.ViewModels.Home
-@using SFA.DAS.PSRService.Web.SubmitActions
-@model SFA.DAS.PSRService.Web.ViewModels.Home.IndexViewModel
+﻿@model SFA.DAS.PSRService.Web.ViewModels.Home.IndexViewModel
 
 @{
     ViewData["Title"] = "Public sector reporting";
+    ViewBag.GaData.Vpv = "/page-home";
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/SFA.DAS.PSRService.Web/Views/Home/Index.cshtml]: what is the format? should it be page-public-sector-reporting-home?

File: src/SFA.DAS.PSRService.Web/Views/Shared/_GoogleTagManager.cshtml [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+                event: 'gtm.js'
+            });
+            var f = d.getElementsByTagName(s)[0],
+                j = d.createElement(s),
+                dl = l !== 'dataLayer' ? '&l=' + l : '';
+            j.async = true;
+            j.src =
+                'https://www.googletagmanager.com/gtm.js?id=' + i + dl;
+            f.parentNode.insertBefore(j, f);
+        })(window, document, 'script', 'dataLayer', 'GTM-KWQBWGJ');
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/SFA.DAS.PSRService.Web/Views/Shared/_GoogleTagManager.cshtml]: 👍

File: src/SFA.DAS.PSRService.Web/Views/Shared/_GoogleTagManagerNoScript.cshtml [hunk 11]
```diff
@@ -0,0 +1,6 @@
+﻿ <!-- Google Tag Manager (noscript) -->
+<noscript>
+    <iframe src="https://www.googletagmanager.com/ns.html?id=GTM-KWQBWGJ"
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/SFA.DAS.PSRService.Web/Views/Shared/_GoogleTagManagerNoScript.cshtml]: 👍

File: src/SFA.DAS.PSRService.Web/Views/Home/Index.cshtml (hunk 9 above)
[the remark this answers is not in the document]
Comment (Stephek) [reply src/SFA.DAS.PSRService.Web/Views/Home/Index.cshtml]: "For virtual pageview value, it must be unique and should be constructed from a unique Body Id and the service subdomain."  (https://docs.google.com/document/d/1S1-t60rzHkDKUzhogXoSvlykpt0dvU4k4LYqOBxIFw4/edit)

**Comment by Stephek** [comment]
You've only updated one of the Vpv values.  The rest should be revisited.  I suggest that you confirm with Steve McCabe what the values should be given that he is the end user.

**Pull request merged**